internal/deb/helpers.go

```diff
@@ -0,0 +1,44 @@
+package deb
+
+import (
+	"fmt"
+	"runtime"
+)
+
+type archPair struct {
+	goArch  string
+	debArch string
+}
+
+var knownArchs = []archPair{
+	{"386", "i386"},
+	{"amd64", "amd64"},
+	{"arm", "armhf"},
+	{"arm64", "arm64"},
+	{"ppc64le", "ppc64el"},
+	{"riscv64", "riscv64"},
+	{"s390x", "s390x"},
+}
+
+var PlatformGoArch = runtime.GOARCH
+
+func inferDebArchFromPlatformGoArch() (string, error) {
+	for _, arch := range knownArchs {
+		if arch.goArch == PlatformGoArch {
+			return arch.debArch, nil
+		}
+	}
+	return "", fmt.Errorf("cannot infer package architecture from current platform architecture: %s", runtime.GOARCH)
+}
+
+func ValidateArch(debArch string) (string, error) {
+	if debArch == "" {
+		return inferDebArchFromPlatformGoArch()
```

Comment:
Can we please have this as two functions, that do what they say on the tin:
```
func ValidateArch(arch string) error
func InferArch() (string, error)
```
An empty architecture should be an error for the first one, and it doesn't need to return the same string is passed in anymore as it's indeed just validating it. Infer, on the other hand, doesn't need any input as it's just inferring.

Having them both as one is also possible, but the merging makes the interface unnatural and makes one ask what's going on instead of it just being obvious.